weblab/entities/views.py

```diff
@@ -778,6 +781,54 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class TransferView(LoginRequiredMixin, UserFormKwargsMixin, UserPassesTestMixin, FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_transfer_ownership.html'
+    context_object_name = 'entity'
+    form_class = EntityTransferForm
+
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_managed_by(self.request.user)
+
+    def post(self, request, *args, **kwargs):
+        """Check the form and possibly rename the entity.
+
+        Called by Django when a form is submitted.
+        """
+        form = self.get_form()
+
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            entity = self.get_object()
+            user = User.objects.filter(email=email).first()
+
+            if user is None:
+                form.add_error(None, "Unknown user")
+                return self.form_invalid(form)
+
+            if self.model.objects.filter(name=entity.name, author=user).exists():
```

Comment:
@jonc125 I wonder if you could help me with this line of code please. It seems to return true even when the other user doesn't have an entity of the same name.

Reply:
I can't see any errors here, and indeed the `test_transfer` is passing on Travis, which suggests it's working as intended? The build failure is just due to flake8 & isort errors - you might want to look at installing and running those locally before commit/push?